artifacts/definitions/Generic/Detection/HashHunter.yaml

```diff
@@ -45,14 +45,23 @@ parameters:
 sources:
   - query: |
       -- setup hash lists
-      LET MD5Array <= split(sep='\\s+',string=MD5List)
-      LET SHA1Array <=  split(sep='\\s+',string=SHA1List)
-      LET SHA256Array <= split(sep='\\s+',string=SHA256List)
+      LET MD5List <= if(condition= MD5List,
+                        then= split(sep='\\s+',string=MD5List), else=Null)
+      LET SHA1List <= if(condition= SHA1List,
+                        then= split(sep='\\s+',string=SHA1List), else=Null)
+      LET SHA256List <= if(condition= SHA256List,
+                        then= split(sep='\\s+',string=SHA256List), else=Null)
+      
+      -- set hash selector for optimised hash calculation
+      LET HashSelector <= SELECT * FROM chain(
+            a={SELECT "MD5" AS Hash FROM scope() WHERE MD5List},
+            b={SELECT "SHA1" AS Hash FROM scope() WHERE SHA1List},
+            c={SELECT "SHA256" AS Hash FROM scope() WHERE SHA256List})
```

Comment:
probably minor but I was suggesting to pre-calculate this so we dont have to run this query for each file.

Reply:
shouldnt LET HashSelector <= SELECT precalculate? I initially left it off then added after realising :)